app/src/main/java/com/iti/android/team1/ebuy/ui/register_screen/viewmodel/RegisterViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package com.iti.android.team1.ebuy.ui.register_screen.viewmodel
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.iti.android.team1.ebuy.model.datasource.repository.IRepository
+import com.iti.android.team1.ebuy.model.networkresponse.NetworkResponse
+import com.iti.android.team1.ebuy.model.pojo.Customer
+import com.iti.android.team1.ebuy.model.pojo.CustomerRegister
+import com.iti.android.team1.ebuy.ui.register_screen.ErrorType
+import com.iti.android.team1.ebuy.ui.register_screen.RegisterResult
+import com.iti.android.team1.ebuy.util.AuthRegex
+import com.iti.android.team1.ebuy.util.Decoder
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+
+class RegisterViewModel(private val repoInterface: IRepository) : ViewModel() {
+
+    private val _registerMutableLiveData: MutableLiveData<RegisterResult?> = MutableLiveData()
+    val registerLiveData = _registerMutableLiveData as LiveData<*>
+
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun registerCustomer(customerRegister: CustomerRegister) {
+
+        _registerMutableLiveData.value = RegisterResult.Loading
+
+        when {
+            isUserNameInvalid(customerRegister.firstName) -> {
+                _registerMutableLiveData.value =
+                    RegisterResult.InvalidData(ErrorType.FirstNameError)
+            }
+            isUserNameInvalid(customerRegister.lastName) -> {
+                _registerMutableLiveData.value =
+                    RegisterResult.InvalidData(ErrorType.LastNameError)
+            }
+            !AuthRegex.isEmailValid(customerRegister.email) -> {
+                _registerMutableLiveData.value =
+                    RegisterResult.InvalidData(ErrorType.EmailError)
+            }
+            !AuthRegex.isPasswordValid(customerRegister.password) -> {
+                _registerMutableLiveData.value =
+                    RegisterResult.InvalidData(ErrorType.PasswordError)
+            }
+            else -> register(customerRegister.copy(password = Decoder.encode(customerRegister.password)))
```

Comment:
I believe the decode and encode are associated with the Repository or RemoteSource Not In ViewModel.

Reply:
i believe that repoitory resposable for get and send data not modify data

Reply:
And modify it for remote source the view model shouldn't know any thing about the remote source